extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -151,6 +163,38 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		}
 	}
 
+	async saveJupyterBooks(): Promise<void> {
+		if (this._bookPath) {
+			const allFilesFilter = localize('allFiles', "All Files");
+			let filter: any = {};
+			filter[allFilesFilter] = '*';
+			let uris = await vscode.window.showOpenDialog({
+				filters: filter,
+				canSelectFiles: false,
+				canSelectMany: false,
+				canSelectFolders: true,
+				openLabel: localize('labelPickFolder', "Pick Folder")
+			});
+			if (uris && uris.length > 0) {
+				let pickedFolder = uris[0];
+				let destinationUri: vscode.Uri = vscode.Uri.file(path.join(pickedFolder.fsPath, path.basename(this._bookPath)));
+				if (destinationUri) {
+					//remove folder if exists
+					await fs.remove(destinationUri.fsPath);
+					//make directory for each contribution book.
+					await fs.mkdirSync(destinationUri.fsPath);
```

Comment:
>mkdirSync [](start = 14, length = 9)

Same here - you're awaiting a sync method

Reply:
mkdir doesn't return a promise - this is doing nothing. You need to promisfy mkdir (and all the other fs calls)

---
In reply to: [322982164](https://github.com/microsoft/azuredatastudio/pull/7037#discussion_r322982164) [](ancestors = 322982164)